### DISCUSSION TITLE
Can't Run Ubuntu 18.04 in UTM

### DISCUSSION THREAD

**Discussion by ejxousiva** [utmapp/UTM#3735] [UTM Help (macOS)]
**Describe the issue**
Both Ubuntu amd, arm 18.04 doesn't work at my UTM. I don't know how to progress. How can I boot up successfully?

**Configuration**
* UTM Version: 3.0.4
* OS Version: Monterey 12.2
* Intel or Apple Silicon? Apple Silicon

**Upload VM**

https://user-images.githubusercontent.com/48287720/157423628-36522e47-5260-4d0b-a4ea-2bd8fcb5a2f0.mov

https://user-images.githubusercontent.com/48287720/157423434-695b9b75-1aa6-4402-a6d9-0a475034cde2.mov

**Comment by osy** [comment]
First one is because you’re using AMD64 not ARM64. Second one you’ve just did not wait long enough.

**Reply by jhgorse** [reply]
He seems to be using the ARM iso in the second video.
<img width="617" alt="image" src="https://user-images.githubusercontent.com/1783474/157502551-1bac760e-00a6-4a97-9e7b-7ef6aa24a4e6.png">

Likewise, I cannot get the ARM64 Ubuntu 18.04 install to boot. I get the boot menu. Select install. Then it hangs with the blinking '_' cursor in the upper left.

Where is the QEMU console and logs?

**Reply by osy** [reply]
Does 20.04 arm64 work?

**Reply by jhgorse** [reply]
@osy yes. Kind of. I had to disable the network in order to avoid an installer partitioning crash:
https://askubuntu.com/questions/1232261/ubuntu-20-04-server-install-fails-during-partitioning-subiquity-failure

It appears to be booting now without network.

**Reply by jhgorse** [reply]
Log output:

```
Running:  -L /Applications/UTM.app/Contents/Resources/qemu -S -qmp tcp:127.0.0.1:4001,server,nowait -nodefaults -vga none -spice "unix=on,addr=/Users/joe.gorse/Library/Group Containers/WDNLXAD4W8.com.utmapp.UTM/5FD9A729-E6FE-448D-8075-D74DB0E07A8B.spice,disable-ticketing=on,image-compression=off,playback-compression=off,streaming-video=off,gl=off" -device virtio-ramfb -device virtio-rng-pci -cpu cortex-a72 -smp cpus=4,sockets=1,cores=4,threads=1 -machine virt,highmem=off -accel tcg,tb-size=2048 -drive if=pflash,format=raw,unit=0,file=/Applications/UTM.app/Contents/Resources/qemu/edk2-aarch64-code.fd,readonly=on -drive "if=pflash,format=raw,unit=1,file=/Users/joe.gorse/Library/Containers/com.utmapp.UTM/Data/Documents/Ubuntu 18.04 ARM64.utm/Images/efi_vars.fd" -boot menu=on -m 8192 -device intel-hda -device hda-duplex -name "Ubuntu 18.04 ARM64" -device nec-usb-xhci,id=usb-bus -device usb-tablet,bus=usb-bus.0 -device usb-mouse,bus=usb-bus.0 -device usb-kbd,bus=usb-bus.0 -device ich9-usb-ehci1,id=usb-controller-0 -device ich9-usb-uhci1,masterbus=usb-controller-0.0,firstport=0,multifunction=on -device ich9-usb-uhci2,masterbus=usb-controller-0.0,firstport=2,multifunction=on -device ich9-usb-uhci3,masterbus=usb-controller-0.0,firstport=4,multifunction=on -chardev spicevmc,name=usbredir,id=usbredirchardev0 -device usb-redir,chardev=usbredirchardev0,id=usbredirdev0,bus=usb-controller-0.0 -chardev spicevmc,name=usbredir,id=usbredirchardev1 -device usb-redir,chardev=usbredirchardev1,id=usbredirdev1,bus=usb-controller-0.0 -chardev spicevmc,name=usbredir,id=usbredirchardev2 -device usb-redir,chardev=usbredirchardev2,id=usbredirdev2,bus=usb-controller-0.0 -device usb-storage,drive=cdrom0,removable=true,bootindex=0,bus=usb-bus.0 -drive if=none,media=cdrom,id=cdrom0 -device virtio-blk-pci,drive=drive0,bootindex=1 -drive "if=none,media=disk,id=drive0,file=/Users/joe.gorse/Library/Containers/com.utmapp.UTM/Data/Documents/Ubuntu 18.04 ARM64.utm/Images/data.qcow2,cache=writethrough" -nic none -device virtio-serial -device virtserialport,chardev=vdagent,name=com.redhat.spice.0 -chardev spicevmc,id=vdagent,debug=0,name=vdagent -device virtserialport,chardev=charchannel1,id=channel1,name=org.spice-space.webdav.0 -chardev spiceport,name=org.spice-space.webdav.0,id=charchannel1 -uuid 5FD9A729-E6FE-448D-8075-D74DB0E07A8B -rtc base=localtime
qemu-aarch64-softmmu: warning: Spice: playback:0 (0x14587c940): setsockopt failed, Operation not supported on socket
qemu-aarch64-softmmu: warning: Spice: record:0 (0x14587c9f0): setsockopt failed, Operation not supported on socket
```

**Reply by jhgorse** [reply]
Disabling Full Graphics in favor of Console only appears to resolve the initial blocker.
<img width="933" alt="image" src="https://user-images.githubusercontent.com/1783474/157512295-c32dbb4e-0730-4fcb-8a29-541a3d3b2f66.png">

**Reply by whatboxapp** [reply]
arm64

**Reply by malhassabi** [reply]
> First one is because you’re using AMD64 not ARM64. Second one you’ve just did not wait long enough.

how long should one have to wait? i let mine sit for more than 10 minutes at a time without any change. 

20.04 on the other hand will post --> select install/try ubuntu --> install menu < 2 min.

**Reply by n4226** [reply]
Did you find the solution? I'm running into the same issue. It is just stuck on that screen for a very long time.

**Reply by ejxousiva** [reply]
No. I just gave up on version 18.

**Reply by iltertaha** [reply]
https://github.com/utmapp/UTM/discussions/4564#discussioncomment-3956313 You can install it by adding a serial device with default configurations as mentioned here. Hope that helps.
